### PR TITLE
Refactorizations & replace Parcel with kirbyup

### PR DESCRIPTION
**Pull request by johannschopplich**
Hey there!

First, let me tell you how great of a plugin it is. Thanks for sharing! I've learned a lot on how to use Mapbox, but also integrating it into the Kirby Panel as well.

I have refactored some parts of your plugin, especially the Vue components. Without changing any of their functionality. There were some bad practices (modifying properties from within a computed instance for examples), which ESLint caught. I have also modernized the code base to use ES2020+ features, which helped us to drop some lines. Of course, in the bundling process, those features will be transpiled to support older ES versions.

A somewhat important feature: The Mapbox scripts and styles will now only be loaded once. Before, they were fetched every time the block field was opened. 

Parcel 1 is no longer supported — [kirbyup](https://getkirby.com/docs/guide/plugins/plugin-setup-panel#introducing-kirbyup) replaces it. (Already part of the new Kirby Panel pluginkit.)

Last but not least, I fixed some grammar and [capitalization](https://en.wikipedia.org/wiki/Capitalization_in_English) issues in your readme. Some conventions from German slipped in there. 

I tried to create as snackable commits as possible. I hope you like the changes! 🙂

### CODE REVIEW

**Review by youngcut — CHANGES_REQUESTED** [review]
I'm glad to see that this plugin has been well liked. I'm a bit proud of it by myself. 😇 
I have to admit, these are my first steps with Vue, Kirby and GitHub, so I'm very happy that someone took a look at my code. It's a way for me to improve myself. Thank you very much  for that. 🙌
I found two lines that I don't think this can work like this.
Otherwise everything looks great.
Might you be able to review [this plugin](https://github.com/youngcut/kirby-grid-block) as well? 🥺

**Comment by johannschopplich** [comment]
Wow, for your first experience with Kirby and Vue, those plugins are quite well written. Seriously. Glad to hear, you are happy about those changes. 🙂 

Sadly, I wasn't able to track down the changes you requested. GitHub didn't showd me conversations or change requests. Are you sure you published the review? Could you please add a review again? Or maybe comment a line? I'm looking forward to take a look at those lines!

Yes, I want to take a look into the other plugin as well. Glad we can work together. 😊

**Comment by johannschopplich** [comment]
Also, an idea for the GitHub description. Feel free to copy:
> 🗺 Geolocation field & map block: All you need for a map on your website

File: src/components/Map.vue [hunk 1]
```diff
@@ -1,197 +1,186 @@
...
+      s.src = `${cdnUrl}/${jsLib}`;
+      document.body.appendChild(s);
+    } else {
+      this.initMap();
+    }
+  },
+
+  methods: {
+    async initMap() {
+      const { maps } = await this.$api.get("katapult/options");
```
Comment (youngcut) [inline src/components/Map.vue:112]: Where is katapult defined?

**Comment by youngcut** [comment]
Like i said. Im a beginner in GitHub. 🙈 
(forgot to apply the comments of the lines  🤦‍♂️). Just a moment...

**Comment by johannschopplich** [comment]
@youngcut I found the issues you mentioned. Good eye! My bad. 🙈 I removed the leftovers from internal testings.

**Review by youngcut — APPROVED** [review]
Perfect. 👌

**Pull request merged**